src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -808,22 +811,36 @@ void ShadingPoint::compute_shading_basis() const
         tangent = m_assembly_instance_transform.vector_to_parent(tangent);
     }
 
-    // Construct an orthonormal basis.
-    const Vector3d& sn = get_original_shading_normal();
-    const Vector3d bitangent = cross(tangent, sn);
-    const double norm_bitangent = norm(bitangent);
-    if (norm_bitangent >= 1.0e-6)
+
+    if (m_primitive_type == PrimitiveCurve3)
     {
-        const Vector3d t = bitangent / norm_bitangent;
-        const Vector3d s = cross(sn, t);
-        m_shading_basis.build(sn, s, t);
+        // Contruct shading basis for hair BSDF.
+        // todo: add flag to differential curves from hair.
```

Comment:
How do curves and hair differ? Also, I think you meant "distinguish"?

Reply:
Yes, sorry that was a typo. I wanted to write differentiate/distinguish. Well, by curves and hair differing, I meant that the shading basis required by the BSDF is different. However, I'm wondering whether it is better for all curves to possess the same shading basis and construct a specific basis within the hair BSDF separately? I'm open to ideas.